boot/bootdata/hivesys.inf

```diff
@@ -2191,6 +2191,13 @@ HKLM,"SYSTEM\CurrentControlSet\Services\fastfat","ImagePath",0x00020000,"system3
 HKLM,"SYSTEM\CurrentControlSet\Services\fastfat","Start",0x00010001,0x00000003
 HKLM,"SYSTEM\CurrentControlSet\Services\fastfat","Type",0x00010001,0x00000002
 
+; Virtual FATX filesystem driver
+HKLM,"SYSTEM\CurrentControlSet\Services\vfatfs","ErrorControl",0x00010001,0x00000001
+HKLM,"SYSTEM\CurrentControlSet\Services\vfatfs","Group",0x00000000,"Boot File System"
+HKLM,"SYSTEM\CurrentControlSet\Services\vfatfs","ImagePath",0x00020000,"system32\drivers\vfatfs.sys"
+HKLM,"SYSTEM\CurrentControlSet\Services\vfatfs","Start",0x00010001,0x00000003
+HKLM,"SYSTEM\CurrentControlSet\Services\vfatfs","Type",0x00010001,0x00000002
```

Comment:
Wouldn't this fail on non-xbox systems ?

Reply:
3 is SERVICE_DEMAND_START. If a FATX partition is detected on non-XBOX system, kernel will try to start and fail to load the driver, and the partition would not be mounted.
If these lines are not present, kernel would fail to locate the vfatfs service announced by fs_rec, and the partition would not be mounted.

Reply:
Yes, those lines should be present like for example, any other filesystem driver supported (fastfat, ntfs, etc.)

Reply:
You might want to keep the registry data in a dedicated .inf file (vfatfs_reg.inf).